### ISSUE TITLE
Error When doing Subquery

### ISSUE THREAD

**Issue by daryllabar** [MarkMpn/Sql4Cds#391]
I'm unable to run this sql, getting an "Object reference not set to an instance of an object" error

``` sql
  SELECT MAX(acme_upcpeergroupsid) AS Id, acme_UPC 
  FROM acme_upcpeergroups
  WHERE acme_peergroupid IS NULL
    AND acme_UPC NOT IN (
      SELECT acme_UPC FROM acme_upcpeergroups WHERE acme_peergroupid = 'bfde2e9d-740e-ee11-8f6c-001dd8094632'
    )
  GROUP BY acme_upc
```

**Comment by MarkMpn** [comment]
I'm struggling to reproduce this. Can you get a similar query using OOB entities that produces the same error, or send me the schema of this table?

**Comment by daryllabar** [comment]
It might be a plugin on the server causing the issues.  I'll close it for now, and reopen if I can give a clean OOB issue.

**Comment by MarkMpn** [comment]
It sounds most likely to be a SQL 4 CDS issue rather than plugin related, but possibly very specific to the schema. If you can't reproduce it with an OOB table but you're able to share a solution that includes this entity definition I'd be happy to take a look.

**Comment by MarkMpn** [comment]
@realpobblebonk Are you able to share the query, or even produce a similar query that results in the same error using standard entities so I can reproduce the problem?

**Comment by realpobblebonk** [comment]
I update my original post with more details including the fact that I can get the query to run in version 7.4.0.0

**Comment by MarkMpn** [comment]
Is it possible for you to share a solution containing these entities so I can try running this query?

**Comment by MarkMpn** [comment]
I believe this may be fixed in the latest version, can you please re-test in 8.0.0?

**Comment by MarkMpn** [comment]
I'm assuming this is now resolved in 8.0.0, please let me know if you're still having the same problem after updating.
